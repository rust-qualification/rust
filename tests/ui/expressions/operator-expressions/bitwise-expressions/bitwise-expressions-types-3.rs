// Bitwise expressions: d7a0044c-a6c7-4a18-8b26-d6f3ff619e7b

pub fn main() {
    let a: bool = true;
    let b: bool = false;
    
    // invalid types
    let a_shift = a << b; 
    //~^ ERROR no implementation for `bool << bool`
    let b_shift = a >> b; 
    //~^ ERROR no implementation for `bool >> bool`
}