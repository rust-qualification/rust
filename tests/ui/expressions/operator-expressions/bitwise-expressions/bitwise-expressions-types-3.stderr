error[E0369]: no implementation for `bool << bool`
  --> $DIR/bitwise-expressions-types-3.rs:8:21
   |
LL |     let a_shift = a << b; 
   |                   - ^^ - bool
   |                   |
   |                   bool

error[E0369]: no implementation for `bool >> bool`
  --> $DIR/bitwise-expressions-types-3.rs:10:21
   |
LL |     let b_shift = a >> b; 
   |                   - ^^ - bool
   |                   |
   |                   bool

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0369`.
